### PR TITLE
gamescope: 3.12.5 -> 3.13.19

### PR DESCRIPTION
**Pull request by Scrumplex**
## Description of changes

Continuation of #270043
Closes #270043

### This package is still outdated

~~At the time of writing the latest version of Gamescope is 3.13.12. But that version uses wlroots 0.17, which requires #269359 to be merged first.~~

### New dependencies

reshade and SPIRV-Headers have been added as new build-time dependencies. The latter is packaged in nixpkgs and is (for now) substituted as a submodule.

The former is not packaged and is fetched from the upstream repository. Note that gamescope uses https://github.com/Joshua-Ashton/reshade/tree/9fdbea6892f9959fdc18095d035976c574b268b7, while we use https://github.com/crosire/reshade/tree/5c8b8f0993a46fcd2f471181b002fb7f549c33c5 which is just the upstream project, with the changes from Joshua-Ashton.

vkroots has also been bumped.

### Future work

It would be nice, if we could use pkg-config for SPIRV-Headers. I'd say this is an opportunity for an upstream contribution.

### All Changes

https://github.com/ValveSoftware/gamescope/compare/3.12.5...3.13.19

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by IogaMaster** [comment]
@Scrumplex Thanks for this! I have been too busy to update the pr.

**Comment by matthewpi** [comment]
Result of `nixpkgs-review pr 271422` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>gamescope</li>
    <li>gamescope.lib</li>
  </ul>
</details>

**Review by PedroHLC — APPROVED** [review]
Ran without issues, didn't test the WSI layer but files are in-place.

**Comment by gaykitty** [comment]
Result of `nixpkgs-review pr 271422` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>gamescope</li>
    <li>gamescope.lib</li>
  </ul>
</details>

**Comment by nrdxp** [comment]
seeing as wlroots_17 has been merged, perhaps we can update this to the latest version now?

**Review by nrdxp — COMMENTED** [review]
Can we just squash this into 1 commit?

File: pkgs/applications/window-managers/gamescope/default.nix [hunk 1]
```diff
@@ -102,15 +110,18 @@ stdenv.mkDerivation {
...
-    vkroots
     libdisplay-info
   ];
 
   outputs = [ "out" "lib" ];
 
   postUnpack = ''
     rm -rf source/subprojects/vkroots
-    ln -s ${vkroots} source/subprojects/vkroots
+    ln -s ${finalAttrs.vkroots} source/subprojects/vkroots
```
Comment (nrdxp) [inline pkgs/applications/window-managers/gamescope/default.nix:120]: Should this be labeled as a `# HACK`? Maybe put in the same comment from the description about possible upstream improvements?

**Review by nrdxp — APPROVED** [review]
This more or less looks good at this point. Not super happy about the hack, but I agree it is probably best addressed upstream. I made a quick go at it, but ran out of time. Hopefully I can swing back to it before too long, but this should do for now.

**Comment by anthr76** [comment]
I've been testing out this package but I'm not seeing https://github.com/ValveSoftware/gamescope/blob/master/layer/VkLayer_FROG_gamescope_wsi.json.in installed

**Comment by Scrumplex** [comment]
> I've been testing out this package but I'm not seeing https://github.com/ValveSoftware/gamescope/blob/master/layer/VkLayer_FROG_gamescope_wsi.json.in installed

That layer should be in `lib` output of this package

**Comment by anthr76** [comment]
Hm I'm not seeing it but may be missing it. Can you confirm? I'm unsure how to build a package in nixpkgs so I can poke through the derivation :( I tried `nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'`

**Comment by Scrumplex** [comment]
> Hm I'm not seeing it but may be missing it. Can you confirm?

I checked yesterday and the `lib` output had the vulkan layer and manifest as expected

> I'm unsure how to build a package in nixpkgs so I can poke through the derivation :( I tried `nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'`

Try either `nix build .#gamescope.lib` (if you have nix-command and flakes enabled) or `nix-build -A gamescope.lib` in the repo

**Comment by anthr76** [comment]
Hm I do see that in my derivation outputs but I'm not seeing it anywhere else in my filesystem besides nixstore (which may be expected). I for example can't seem to get VK_LOADER logs to tell me anything about it invoking with `env VK_LOADER_DEBUG=error,warn,info ENABLE_HDR_WSI=1 ENABLE_GAMESCOPE_WSI=1 gamescope --hdr-enabled -- env QML2_IMPORT_PATH="" moonlight`

Here's a gist of the logs https://gist.github.com/anthr76/6d4b3abff90118d6b90bc0b1974bbe36

I don't see any mention of VkLayer_FROG_gamescope_wsi.

Thanks a lot for your help

**Comment by Scrumplex** [comment]
> I don't see any mention of VkLayer_FROG_gamescope_wsi.

The original reason for this is here: https://github.com/NixOS/nixpkgs/pull/255293

I guess we all just forgot to continue from there ^^

**Comment by PedroHLC** [comment]
TL;DR Vulkan stack uses `XDG_DATA_DIRS` for searching layers, which includes systemPackages, but our NixOS module for gamescope replaces `.out` in systemPackages with a wrapper with args+envs and/or a wrapper with sec bits. Having the layer in `.lib` allows you to address this without any chance of conflict or fuzz.

**Comment by Scrumplex** [comment]
Closing in favor of #280465
